src/dotnet/commands/dotnet-restore3/Restore3CommandApp.cs

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using Microsoft.DotNet.Cli;
+using Microsoft.DotNet.Cli.CommandLine;
+
+namespace Microsoft.DotNet.Tools.Restore3
+{
+    public class Restore3CommandApp
+    {
+        public static int Execute(string[] args)
+        {
+            CommandLineApplication cmd = new CommandLineApplication
+            {
+                Name = "restore3",
+                FullName = "restore3",
+                Description = "restore for msbuild"
+            };
+
+            cmd.HelpOption("-h|--help");
+
+            var argRoot = cmd.Argument(
```

Comment:
Do we want this, or do we want to support 'pass through' args?  like "dotnet run" does.  Basically allow the user to pass any arg through to the MSBuild process.

To allow this, you need to set `throwOnUnexpectedArg` = false, and grab the RemainingArguments from the CommandLineApplication.